prow/cmd/tackle/ingress.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	extensions "k8s.io/api/extensions/v1beta1"
+	networking "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+)
+
+var runtimeScheme = runtime.NewScheme()
+
+func init() {
+	extensions.AddToScheme(runtimeScheme)
+	networking.AddToScheme(runtimeScheme)
+}
+
+// networkingIngressAvailable checks if the "networking.k8s.io/v1beta1" resource is available based on Kubernetes version.
```

Comment:
Not entirely convinced any of these 1-3 line functions need to exist (they are only called by one function).

Not entirely convinced this file needs to exist

Reply:
> Not entirely convinced any of these 1-3 line functions need to exist (they are only called by one function).

I do not have a problem killing the function since it may be overkill for a single expression.

> Not entirely convinced this file needs to exist

Are you not convinced because you feel the logic to do this should already be available in `apimachinery` or because you prefer this to be in [`main.go`](https://github.com/kubernetes/test-infra/pull/13597/files#diff-85376b10136f7ee03443261541015cf2)